mono/mini/interp/interp.c

```diff
@@ -390,8 +390,9 @@ get_virtual_method (InterpMethod *imethod, MonoObject *obj)
 }
 
 static void inline
-stackval_from_data (MonoType *type, stackval *result, char *data, gboolean pinvoke)
+stackval_from_data (MonoType *type_, stackval *result, char *data, gboolean pinvoke)
 {
+	MonoType *type = mini_native_type_replace_type (type_);
```

Comment:
We would need to audit a lot of usages and it always makes sense to replace it. Also, we do something similar in `mini`: https://github.com/mono/mono/tree/555041c8c2d5db324e3cb79fb65e614a4e347f64/mono/mini/mini-generic-sharing.c#L3258